sklearn/decomposition/tests/test_nmf.py

```diff
@@ -502,10 +503,10 @@ def test_nmf_decreasing():
             previous_loss = None
             for n_iter in range(1, 5):
                 W, H, n_iter = non_negative_factorization(
-                    X, W, H, beta_loss=beta_loss, n_components=n_components,
-                    max_iter=n_iter, alpha=alpha, solver=solver, tol=tol,
-                    l1_ratio=l1_ratio, verbose=0, regularization='both',
-                    random_state=0, update_H=True)
+                    X, W0.copy(), H0.copy(), beta_loss=beta_loss,
+                    n_components=n_components, max_iter=n_iter, alpha=alpha,
+                    solver=solver, tol=tol, l1_ratio=l1_ratio, verbose=0,
+                    regularization='both', random_state=0, update_H=True)
```

Comment:
I would instead insert:

```python
W, H = W0.copy(), H0.copy()
previous_loss = None
for n_iter in range(1, 5):
    W, H, n_iter = non_negative_factorization(X, W, H, ...)
```

so as to check the monotonicty insude the `n_iter` loop.